tools/ci-operator-prowgen/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/ghodss/yaml"
+	cioperatorapi "github.com/openshift/ci-operator/pkg/api"
+	"io/ioutil"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowkube "k8s.io/test-infra/prow/kube"
+	"os"
+	"path"
+	"strings"
+)
+
+type options struct {
+	ciOperatorConfigPath string
+
+	help    bool
+	verbose bool
+}
+
+func bindOptions(flag *flag.FlagSet) *options {
+	opt := &options{}
+
+	flag.StringVar(&opt.ciOperatorConfigPath, "source-config", "", "Path to ci-operator configuration file in openshift/release repository.")
+	flag.BoolVar(&opt.help, "h", false, "Show help for ci-operator-prowgen")
+	flag.BoolVar(&opt.verbose, "v", false, "Show verbose output")
+
+	return opt
+}
+
+func generatePodSpec() *prowkube.PodSpec {
+	podspec := prowkube.PodSpec{ServiceAccountName: "ci-operator"}
+	command := []string{"ci-operator"}
+	args := []string{"--artifact-dir=$(ARTIFACTS)"}
+
+	// TODO: prowkube has no EnvVarSource alias
+	// configMapKeyRef := prowkube.EnvVarSource{ConfigMapKeyRef: &prowkube.ConfigMapKeySelector{LocalObjectReference: prowkube.LocalObjectReference{Name: fmt.Sprintf("ci-operator-%s-%s", org, component)}, Key: fmt.Sprintf("%s.json", branch)}}
+	configSpec := prowkube.EnvVar{Name: "CONFIG_SPEC"}
+	env := []prowkube.EnvVar{configSpec}
+
+	container := prowkube.Container{Name: "test", Image: "ci-operator:latest", Command: command, Args: args, Env: env}
+	podspec.Containers = []prowkube.Container{container}
+
+	return &podspec
+}
+
+func generatePresubmitForTest(test, org, component, branch string) prowconfig.Presubmit {
+	presubmit := prowconfig.Presubmit{Agent: "kubernetes", AlwaysRun: true}
+	presubmit.Name = fmt.Sprintf("ci-operator-%s-%s-%s", org, component, test)
+	presubmit.Context = fmt.Sprintf("ci/prow/%s", test)
+	presubmit.RerunCommand = fmt.Sprintf("/test %s", test)
+	presubmit.Trigger = fmt.Sprintf("((?m)^/test( all| %s),?(\\\\s+|$))", test)
+	presubmit.Brancher = prowconfig.Brancher{}
+	presubmit.Brancher.Branches = []string{branch}
+	presubmit.UtilityConfig = prowconfig.UtilityConfig{Decorate: true}
+	// TODO: `skip_cloning` does not seem to be covered by `Presubmit`
+	presubmit.Spec = generatePodSpec()
+
+	return presubmit
+}
+
+func generatePresubmits(tests []cioperatorapi.TestStepConfiguration, org, component, branch string) map[string][]prowconfig.Presubmit {
+	repo := fmt.Sprintf("%s/%s", org, component)
+	presubmits := make(map[string][]prowconfig.Presubmit)
```

Comment:
nit: don't need to explicitly `make` in Go (often), this could be:

```go
presubmits := map[string][]prowconfig.Presubmit{}
for _, test := range tests {
    presubmits[repo] = append(presubmits[repo], ...)
}
```